codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Headers.java

```diff
@@ -113,7 +120,197 @@ public static boolean isPseudoHeader(CharSequence header) {
+                return AUTHORITY;
+            }
+            return null;
+        }
+
+        private static PseudoHeaderName getProtocolIfEqualsNoPrefix(byte[] array, int offset) {
+            if (array[offset + 1] == 'p' && array[offset + 2] == 'r' && array[offset + 3] == 'o' &&
+                array[offset + 4] == 't' && array[offset + 5] == 'o' && array[offset + 6] == 'c' &&
+                array[offset + 7] == 'o' && array[offset + 8] == 'l') {
+                return PROTOCOL;
+            }
+            return null;
+        }
+
+        /**
+         * Returns the {@link PseudoHeaderName} corresponding to the specified header name.
+         *
+         * @return corresponding {@link PseudoHeaderName} if any, {@code null} otherwise.
+         */
+        public static PseudoHeaderName getPseudoHeader(AsciiString header) {
+            int length = header.length();
+            if (length == 0) {
+                return null;
+            }
+            byte[] array = header.array();
+            int offset = header.arrayOffset();
+            if (array[offset] == PSEUDO_HEADER_PREFIX_BYTE) {
+                switch (length) {
+                case 5:
+                    if (header == PATH.value()) {
+                        return PATH;
+                    }
+                    // :path
+                    return getPathIfEqualsNoPrefix(array, offset);
+                case 7:
+                    if (header == METHOD.value()) {
+                        return METHOD;
+                    }
+                    if (header == SCHEME.value()) {
+                        return SCHEME;
+                    }
+                    if (header == STATUS.value()) {
+                        return STATUS;
+                    }
+                    // :method, :scheme, :status
+                    return getMethodSchemeStatusIfEqualsNoPrefix(array, offset);
+                case 9:
+                    if (header == PROTOCOL.value()) {
+                        return PROTOCOL;
+                    }
+                    // :protocol
+                    return getProtocolIfEqualsNoPrefix(array, offset);
+                case 10:
+                    if (header == AUTHORITY.value()) {
+                        return AUTHORITY;
+                    }
+                    // :authority
+                    return getAuthorityIfEqualsNoPrefix(array, offset);
+                }
+            }
+            return null;
+        }
+        private static final int METHOD_STRING_HASHCODE = -1141949029;
+        private static final int SCHEME_STRING_HASHCODE = -972381601;
+        private static final int STATUS_STRING_HASHCODE = -956875604;
+        private static final int PATH_STRING_HASHCODE = 56997727;
+        private static final int AUTHORITY_STRING_HASHCODE = -1332238263;
+        private static final int PROTOCOL_STRING_HASHCODE = 1552659666;
```

Comment:
Why not get these values in a static block instead of hard coding these ?

Reply:
Javac need to see then as constant, in order to use them in the switch cases :/ 
But String::hashCode is a fixed contractor, hence shouldn't be a big deal

Reply:
But is ugly, nonetheless:"(

Reply:
what's your final thought on this bud?

Reply:
ok I think its fine... maybe add some more comments ?